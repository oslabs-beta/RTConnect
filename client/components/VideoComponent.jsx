import React from 'react'


const VideoComponent = ({handleCreateRoomClick}) => {
  // const handleClick = async () => {
  //   try {
  //     // get local webcam permissions
  //     const myWebCam = await navigator.mediaDevices.getUserMedia({'video': true, 'audio': true});
  //     console.log('Got MediaStream:', myWebcam);
  //     myWebcam.getTracks().forEach((track) => console.log(track))

  //     // set video source to the local stream (myWebCam)
  //     const videoElement = document.querySelector('.localVideo');
  //     videoElement.srcObject = myWebCam;

  //   } catch (error) {
  //     console.error('Error accessing media devices.', error);
  //   }
  // }

  return (
    <div>
      <div>
        Video Component
      </div>
      
      <button onClick={handleCreateRoomClick}>Create Room</button>
      <input type='text'></input>

      {/* <button onClick={handleClick}>Join Room</button>
      <input type='text'></input> */}

      <video className="localVideo" autoPlay playsInline controls={false}/>

    </div>
  )
}

export default VideoComponent

