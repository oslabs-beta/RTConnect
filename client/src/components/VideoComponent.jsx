import React from 'react'
import { useState } from 'react'


const VideoComponent = ({ handleCreateRoomClick, hasJoined, handleJoinRoomClick, joinRoom }) => {
  // const [hasJoined, setHasJoined] = useState(true)
  // const handleClick = async () => {
  //   try {
  //     // get local webcam permissions
  //     const myWebCam = await navigator.mediaDevices.getUserMedia({'video': true, 'audio': true});
  //     console.log('Got MediaStream:', myWebcam);
  //     myWebcam.getTracks().forEach((track) => console.log(track))

  //     // set video source to the local stream (myWebCam)
  //     const videoElement = document.querySelector('.localVideo');
  //     videoElement.srcObject = myWebCam;

  //   } catch (error) {
  //     console.error('Error accessing media devices.', error);
  //   }
  // }

return (
  <div>
    <div style={{display: 'flex', justifyContent: 'space-around', border: '1px solid black'}}>
      <button onClick={handleCreateRoomClick}>Create Room</button>
      <p className='createRoomText'></p>
      <button onClick={joinRoom}>Join Room</button>
      <input type='text'></input>
    </div>
    <div style={{display: 'flex', justifyContent: 'center'}}>
      <video className="localVideo" autoPlay playsInline controls={true}/>
      <video className="remoteVideo" autoPlay playsInline controls={true}/>
    </div>
  </div>
)
  // return (

  //   !hasJoined ? 

  //   <div>
  //     <div style={{display: 'flex', justifyContent: 'space-around', border: '1px solid black'}}>
  //       <button onClick={handleCreateRoomClick}>Create Room</button>
  //       <p className='createRoomText'></p>

  //       <button onClick={() => {hasJoined = !hasJoined}}>Join Room</button>
  //     </div>


  //   <div style={{display: 'flex', justifyContent: 'center'}}>
  //     <video className="localVideo" autoPlay playsInline controls={true} width='80%'/>
      
  //   </div>


  //   </div>
  //   /********************TENERARY********************/ 
  //   :
  //   <div>
  //     <div style={{display: 'flex', justifyContent: 'space-around', border: '1px solid black'}}>
  //       <button onClick={handleCreateRoomClick}>Create Room</button>
  //       <p className='createRoomText'></p>
  //     </div>

  //     <button onClick={handleClick}>Join Room</button>
  //     <input type='text'></input>
  //   <div style={{display: 'flex', justifyContent: 'center'}}>
  //     <video className="localVideo" autoPlay playsInline controls={true}/>
  //     <video className="remoteVideo" autoPlay playsInline controls={true}/>
  //   </div>
  //   </div>

  // )
  
}

export default VideoComponent

