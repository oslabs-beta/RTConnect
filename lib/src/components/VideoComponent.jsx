import React, { useState } from 'react';

const VideoComponent = ({ video }) => {

  return (
    <>
      <video autoPlay playsInline controls={true} ref={video} style={{ width: "500px", height: '500px' }}/>
    </>
    )
  }

export default VideoComponent;
