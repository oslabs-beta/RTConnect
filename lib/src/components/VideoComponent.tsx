import React from 'react';

type videoComponent = {
  video: React.LegacyRef<HTMLVideoElement>
  mediaOptions: { controls: boolean, style: { width: string, height: string }}
}

/**
 * @param {HTMLVideoElement} video 
 * @returns a video component that is either the local or remote video stream
 * controls can be set to true to pause and adjust volumes of streams
 */
const VideoComponent = ({ video, mediaOptions = { controls: true, style: { width: '640px', height: '360px'}}}: videoComponent) => {
  const { controls, style } = mediaOptions;

  return (
    <>
      <video 
        className="videoPlayer" 
        autoPlay
        playsInline
        controls={controls}
        ref={video}
        style={style}
      />
    </>
  );
};

export default VideoComponent;
