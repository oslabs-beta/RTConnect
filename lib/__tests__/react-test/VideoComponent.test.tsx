/* eslint-disable @typescript-eslint/no-unused-vars */
/** @jest-environment jsdom */

import React from 'react';
import { render, screen } from '@testing-library/react';
import VideoComponent from '../../src/components/VideoComponent';

describe('VideoComponent', () => {
  // const localVideo = null;
  
  const options = { controls: true, style: { width: '640px', height: '360px'}}; 
  const { controls, style } = options;

  it('controls should be true', () => {
    expect(controls).toBe(true);
    expect(style).toStrictEqual({ width: '640px', height: '360px'});
  });

  // it('render video', () => {
  //   const { container } = render(<VideoComponent video = {localVideo} mediaOptions = {options}/>);
  //   expect(container).toMatchSnapshot();
  // });

});

