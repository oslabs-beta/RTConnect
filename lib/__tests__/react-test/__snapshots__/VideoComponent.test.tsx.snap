// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VideoComponent render video components 1`] = `
<div>
  <video
    autoplay=""
    class="videoPlayer"
    controls=""
    playsinline=""
    style="width: 640px; height: 360px;"
  />
</div>
`;
